Wifi 5ghz enhancement

salvatore-coppola opened the pull request.

salvatore-coppola committed:
Added new properties.

salvatore-coppola committed:
Added test on hex string.

salvatore-coppola committed:
ReplaceD a jre method invokation with an internal one to get compatibility with JRE > 8

salvatore-coppola committed:
Added ac radio mode. Enhanced channels parser. Added automatic channel to web ui

salvatore-coppola committed:
Added ac radio mode. Enhanced channels parser. Added automatic channel to web ui

salvatore-coppola committed:
splitted in regex no ir from radar detection.

salvatore-coppola committed:
Replaced else if chain with a switch case

salvatore-coppola committed:
Added automatic channel bind to 0 channel.

salvatore-coppola committed:
Added automatic channel bind to 0 channel.

salvatore-coppola committed:
Serveral fixes and improves.

salvatore-coppola committed:
Serveral fixes and improves.

salvatore-coppola committed:
Now country code is properly returned back.

salvatore-coppola committed:
Now countrycode i retrieved by the system in HostapdConfigWriter.

salvatore-coppola committed:
Now countrycode i retrieved by the system in HostapdConfigWriter.

salvatore-coppola committed:
Added country code to wpa supplicant.

salvatore-coppola committed:
Several UI fixes. Enhanced channel discovery.

salvatore-coppola committed:
Several ui fixes.
Fixed bug in hostapd writer.

salvatore-coppola committed:
Fixed unit tests.

salvatore-coppola committed:
Fixed possible catastrophic backtracking

salvatore-coppola committed:
Further regex fix

salvatore-coppola committed:
Fixed tests

salvatore-coppola committed:
Fixed possible catastrophic backtracking#2

salvatore-coppola committed:
Fixed possible catastrophic backtracking#3

salvatore-coppola committed:
Fixed possible catastrophic backtracking#4

salvatore-coppola committed:
Removed useless capture group.

salvatore-coppola committed:
Fixed possible catastrophic backtracking#5

MMaiero commented on kura/org.eclipse.kura.api/META-INF/MANIFEST.MF:
```diff
@@ -2,7 +2,7 @@ Manifest-Version: 1.0
 Bundle-Name: org.eclipse.kura.api
 Bundle-SymbolicName: org.eclipse.kura.api;singleton:=true
-Bundle-Version: 2.3.0.qualifier
+Bundle-Version: 3.0.0.qualifier
```
This is for a 5.1 so we are not supposed to make breaking changes. Can we work them out to be still compatible with the old and maybe deprecate it, while having the new API?

salvatore-coppola replied:
At first I only added a new radiomode in [kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/net/wifi/WifiRadioMode.java](WifiRadioMode) but SonarCloud gave me issues with constants name. So I needed to change all enum items in order to complay Sonar request.

MMaiero replied:
Got it, but we cannot make a breaking change. So maybe it is possible to leave the existing there and deprecate them. Adding new and correct ones with a since to specify that they have been added in this new release of the api. Or something similar

salvatore-coppola commented:
Oki!

Da: Matteo Maiero ***@***.***>
Inviato: mercoledì 27 ottobre 2021 09:18
A: eclipse/kura ***@***.***>
Cc: Coppola, Salvatore ***@***.***>; Author ***@***.***>
Oggetto: Re: [eclipse/kura] Wifi 5ghz enhancement (PR #3661)


@MMaiero commented on this pull request.

________________________________

In kura/org.eclipse.kura.api/META-INF/MANIFEST.MF<https://github.com/eclipse/kura/pull/3661#discussion_r737174272>:

> @@ -2,7 +2,7 @@ Manifest-Version: 1.0

 Bundle-ManifestVersion: 2

 Bundle-Name: org.eclipse.kura.api

 Bundle-SymbolicName: org.eclipse.kura.api;singleton:=true

-Bundle-Version: 2.3.0.qualifier

+Bundle-Version: 3.0.0.qualifier

Got it, but we cannot make a breaking change. So maybe it is possible to leave the existing there and deprecate them. Adding new and correct ones with a since to specify that they have been added in this new release of the api. Or something similar

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/eclipse/kura/pull/3661#discussion_r737174272>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AT5XWXFOO3PNGBSB7TFI2SLUI6YU5ANCNFSM5GK7WIIQ>.
Triage notifications on the go with GitHub Mobile for iOS<https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android<https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.

salvatore-coppola committed:
Upgraded packages version.

salvatore-coppola committed:
Added new wifimode in hostapd reader.

MMaiero commented on kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/net/wifi/WifiRadioMode.java:
```diff
@@ -22,7 +22,8 @@
     RADIO_MODE_80211nHT40below(0x04),
-    RADIO_MODE_80211nHT40above(0x05);
+    RADIO_MODE_80211nHT40above(0x05),
+    RADIO_MODE_80211ac(0x06);
```
Does the issue with sonar disappear if this new entry is set with the proper format?

MMaiero replied:
Requires:
```
    /**
     * @since 2.3
     */
```

salvatore-coppola commented:
Shouldn’t as I recall, but I can be wrong.

Da: Matteo Maiero ***@***.***>
Inviato: venerdì 29 ottobre 2021 14:20
A: eclipse/kura ***@***.***>
Cc: Coppola, Salvatore ***@***.***>; Author ***@***.***>
Oggetto: Re: [eclipse/kura] Wifi 5ghz enhancement (PR #3661)


@MMaiero commented on this pull request.

________________________________

In kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/net/wifi/WifiRadioMode.java<https://github.com/eclipse/kura/pull/3661#discussion_r739185224>:

> @@ -22,7 +22,8 @@

     RADIO_MODE_80211g(0x02),

     RADIO_MODE_80211nHT20(0x03),

     RADIO_MODE_80211nHT40below(0x04),

-    RADIO_MODE_80211nHT40above(0x05);

+    RADIO_MODE_80211nHT40above(0x05),

+    RADIO_MODE_80211ac(0x06);

Does the issue with sonar disappear if this new entry is set with the proper format?

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/eclipse/kura/pull/3661#pullrequestreview-792969473>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AT5XWXA5OXUNOLBFB2RZDWTUJKNRPANCNFSM5GK7WIIQ>.
Triage notifications on the go with GitHub Mobile for iOS<https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android<https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.

MMaiero commented on kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/net/wifi/WifiChannel.java:
```diff
@@ -49,38 +61,63 @@ public void setFrequency(Integer frequency) {
-        sb.append("channel:").append(channel).append(", frequency:").append(frequency).append(" MHz");
-        sb.append("]");
-        return sb.toString();
+    public Boolean isNoInitiatingRadiation() {
```
Requires:
```
    /**
     * @since 2.3
     */
```

MMaiero commented on kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/net/wifi/WifiChannel.java:
```diff
@@ -13,21 +13,33 @@
     /** Wifi frequency in MHz **/
     private Integer frequency;
 
+    private Float attenuation;
```
Just informative. Any particular reason in using the Float object instead of the primitive type?

salvatore-coppola replied:
No specific reason, i used boxed type following the other field choice.

MMaiero commented on kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/client/ui/network/TabWirelessUi.java:
```diff
@@ -1377,7 +1425,11 @@ public void onSuccess(List<GwtWifiHotspotEntry> result) {
     private int getSelectedChannelIndex(int channelValue) {
-        for (int i = 0; i < this.channelList.getItemCount(); i++) {
+
+        if (channelValue == 0)
```
Kura Formatting not applied

MMaiero commented on kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/client/ui/network/TabWirelessUi.java:
```diff
@@ -1481,9 +1534,29 @@ private GwtWifiConfig getGwtWifiConfig() {
         gwtWifiConfig.setIgnoreSSID(this.radio3.getValue());
 
+        // Country Code is not editable at moment.
+        gwtWifiConfig.setCountryCode(this.countryCode.getValue());
```
Do we have an item in the issues to make this editable? Otherwise it will be missed

salvatore-coppola replied:
We decided to only read from the system, however in that point i need to return the code from frontend to backend.

MMaiero reviewed (commented):
I believe for most of the files the copyright header needs to be updated to 2021

salvatore-coppola committed:
Ac mode now doesn't display 2.4 ghz channels.
Chanend Wifimode AC constant name.
Added @since on new methods.
Updated license plates.

MMaiero commented:
On a raspberry pi 3, trying to enable ac is possible, but getting error:
```
Nov 01 12:28:17 raspberrypi hostapd[5373]: wlan0: IEEE 802.11 Hardware does not support configured mode (2) (hw_mode in hostapd.conf)
Nov 01 12:28:17 raspberrypi Kura[709]: failed to start hostapd for the wlan0 interface for unknown reason - errorCode=1
Nov 01 12:28:17 raspberrypi Kura[709]: monitor() :: Error enabling wlan0 interface, will try to reset wifi
Nov 01 12:28:24 raspberrypi Kura[709]: Failed to load skin resource, Resource File /skin.js does not exist
Nov 01 12:28:24 raspberrypi Kura[709]: Failed to load skin resource, Resource File /skin.js does not exist
Nov 01 12:28:25 raspberrypi Kura[709]: {entrypoint=WebConsole, identity=admin, ip=192.168.1.161, web.path=/admin/denali/component, rpc.method=GwtComponentServiceImpl.findComponentConfigurations, session.id=node0kjts859jdzpt1k2vbhq53cqkm0} ConfigurationService - Success - Get component configurations: (!(|(service.pid=*SystemPropertiesService)(service.pid=*NetworkAdminService)(service.pid=*NetworkConfigurationService)(service.pid=*SslManagerService)(service.pid=*FirewallConfigurationService)(service.pid=*WireGraphService)(objectClass=org.eclipse.kura.wire.WireComponent)(objectClass=org.eclipse.kura.driver.Driver)(kura.ui.service.hide=true)))
Nov 01 12:28:28 raspberrypi Kura[709]: Get Wifi Country Code impl
Nov 01 12:28:28 raspberrypi Kura[709]: Get Wifi Country Code Output = global
Nov 01 12:28:28 raspberrypi Kura[709]: Get Wifi Country Code Output = country 00: DFS-UNSET
Nov 01 12:28:28 raspberrypi Kura[709]: {entrypoint=WebConsole, identity=admin, ip=192.168.1.161, web.path=/admin/denali/network, rpc.method=GwtNetworkServiceImpl.getWifiCountryCode, session.id=node0kjts859jdzpt1k2vbhq53cqkm0} UI Network - Success - Get Wifi Country Code
Nov 01 12:28:28 raspberrypi Kura[709]: bringing interface wlan0 up
Nov 01 12:28:28 raspberrypi Kura[709]: starting hostapd for the wlan0 interface --> [hostapd, -B, /etc/hostapd-wlan0.conf]
Nov 01 12:28:28 raspberrypi hostapd[5684]: wlan0: IEEE 802.11 Hardware does not support configured mode (2) (hw_mode in hostapd.conf)
Nov 01 12:28:28 raspberrypi Kura[709]: failed to start hostapd for the wlan0 interface for unknown reason - errorCode=1
Nov 01 12:28:28 raspberrypi Kura[709]: monitor() :: Error enabling wlan0 interface, will try to reset wifi
Nov 01 12:28:29 raspberrypi Kura[709]: {entrypoint=WebConsole, identity=admin, ip=192.168.1.161, web.path=/admin/denali/network, rpc.method=GwtNetworkServiceImpl.findFrequencies, session.id=node0kjts859jdzpt1k2vbhq53cqkm0} UI Network - Success - Get Wifi channels and frequencies
Nov 01 12:28:35 raspberrypi Kura[709]: {entrypoint=WebConsole, identity=admin, ip=192.168.1.161, web.path=/admin/denali/network, rpc.method=GwtNetworkServiceImpl.findFrequencies, session.id=node0kjts859jdzpt1k2vbhq53cqkm0} UI Network - Success - Get Wifi channels and frequencies
Nov 01 12:28:40 raspberrypi Kura[709]: bringing interface wlan0 up
Nov 01 12:28:40 raspberrypi Kura[709]: starting hostapd for the wlan0 interface --> [hostapd, -B, /etc/hostapd-wlan0.conf]
Nov 01 12:28:40 raspberrypi hostapd[5883]: wlan0: IEEE 802.11 Hardware does not support configured mode (2) (hw_mode in hostapd.conf)
Nov 01 12:28:40 raspberrypi Kura[709]: failed to start hostapd for the wlan0 interface for unknown reason - errorCode=1
Nov 01 12:28:40 raspberrypi Kura[709]: monitor() :: Error enabling wlan0 interface, will try to reset wifi
Nov 01 12:28:43 raspberrypi Kura[709]: This invalid DhcpServerCfgIP4 configuration is ignored - org.eclipse.kura.net.dhcp.DhcpServerCfg: [ifaceName=eth0, enabled?=false, defaultLeaseTime=7200, maximumLeaseTime=7200, passDNS?=false], org.eclipse.kura.net.dhcp.DhcpServerCfgIP4: [subnet=192.168.1.0, subnetMask=255.255.255.0, prefix=24, routerAddress=192.168.1.41, rangeStart=172.16.0.100, rangeEnd=172.16.0.110, dnsServer=192.168.1.41]
```

For what I remember raspberry pi 3 does not have 5ghz, so it should not possible to select that out of the list.

MMaiero commented:
Setting then the config to n/b/g, it seems to have better success starting hosted, but then it doesn't seem to work properly:

```
Nov 01 12:28:43 raspberrypi Kura[709]: Not rewriting network interfaces file because it is the same
Nov 01 12:28:43 raspberrypi Kura[709]: Get Wifi Country Code Output = global
Nov 01 12:28:43 raspberrypi Kura[709]: Get Wifi Country Code Output = country 00: DFS-UNSET
Nov 01 12:28:43 raspberrypi Kura[709]: Get Wifi Country Code Output = global
Nov 01 12:28:43 raspberrypi Kura[709]: Get Wifi Country Code Output = country 00: DFS-UNSET
Nov 01 12:28:43 raspberrypi Kura[709]: Not rewriting DHCP config file for wlan0 because it is the same
Nov 01 12:28:44 raspberrypi Kura[709]: Writing snapshot - Getting component configurations...
Nov 01 12:28:44 raspberrypi Kura[709]: monitor() :: Found a new network configuration, will check if wifi has been reconfigured ...
Nov 01 12:28:44 raspberrypi Kura[709]: WifiMonitor: configuration for wlan0 has changed
Nov 01 12:28:45 raspberrypi Kura[709]: monitor() :: reload wlan0 kernel module for WiFi mode MASTER
Nov 01 12:28:45 raspberrypi Kura[709]: monitor() :: reload wlan0 using kernel module for WiFi mode MASTER
Nov 01 12:28:45 raspberrypi Kura[709]: Loading init configurations from: 1635769685082...
Nov 01 12:28:45 raspberrypi Kura[709]: bringing interface wlan0 up
Nov 01 12:28:45 raspberrypi Kura[709]: starting hostapd for the wlan0 interface --> [hostapd, -B, /etc/hostapd-wlan0.conf]
Nov 01 12:28:45 raspberrypi Kura[709]: Writing snapshot - Saving /opt/eurotech/Kura/user/snapshots/snapshot_1635769725824.xml...
Nov 01 12:28:46 raspberrypi Kura[709]: Writing snapshot - Saving /opt/eurotech/Kura/user/snapshots/snapshot_1635769725824.xml... Done.
Nov 01 12:28:46 raspberrypi Kura[709]: {entrypoint=WebConsole, identity=admin, ip=192.168.1.161, web.path=/admin/denali/network, rpc.method=GwtNetworkServiceImpl.updateNetInterfaceConfigurations, session.id=node0kjts859jdzpt1k2vbhq53cqkm0} ConfigurationService - Success - Take snapshot
Nov 01 12:28:46 raspberrypi Kura[709]: {entrypoint=WebConsole, identity=admin, ip=192.168.1.161, web.path=/admin/denali/network, rpc.method=GwtNetworkServiceImpl.updateNetInterfaceConfigurations, session.id=node0kjts859jdzpt1k2vbhq53cqkm0} UI Network - Success - Update network interface configurations
Nov 01 12:28:46 raspberrypi Kura[709]: {entrypoint=WebConsole, identity=admin, ip=192.168.1.161, web.path=/admin/denali/network, rpc.method=GwtNetworkServiceImpl.findFrequencies, session.id=node0kjts859jdzpt1k2vbhq53cqkm0} UI Network - Success - Get Wifi channels and frequencies
Nov 01 12:28:47 raspberrypi dhcpd[6994]: Internet Systems Consortium DHCP Server 4.4.1
Nov 01 12:28:47 raspberrypi dhcpd[6994]: Copyright 2004-2018 Internet Systems Consortium.
Nov 01 12:28:47 raspberrypi dhcpd[6994]: All rights reserved.
Nov 01 12:28:47 raspberrypi dhcpd[6994]: For info, please visit https://www.isc.org/software/dhcp/
Nov 01 12:28:47 raspberrypi dhcpd[6994]: Config file: /etc/dhcpd-wlan0.conf
Nov 01 12:28:47 raspberrypi dhcpd[6994]: Database file: /var/lib/dhcp/dhcpd.leases
Nov 01 12:28:47 raspberrypi dhcpd[6994]: PID file: /var/run/dhcpd-wlan0.pid
Nov 01 12:28:47 raspberrypi dhcpd[6994]: Wrote 0 leases to leases file.
Nov 01 12:28:47 raspberrypi dhcpd[6994]: Listening on LPF/wlan0/b8:27:eb:28:eb:36/172.16.1.0/24
Nov 01 12:28:47 raspberrypi dhcpd[6994]: Sending on   LPF/wlan0/b8:27:eb:28:eb:36/172.16.1.0/24
Nov 01 12:28:47 raspberrypi dhcpd[6994]: Sending on   Socket/fallback/fallback-net
Nov 01 12:28:47 raspberrypi dhcpd[6994]: Server starting service.
Nov 01 12:28:57 raspberrypi Kura[709]: bringing interface wlan0 up
Nov 01 12:28:58 raspberrypi dhcpd[6994]: receive_packet failed on wlan0: Network is down
Nov 01 12:28:58 raspberrypi Kura[709]: starting hostapd for the wlan0 interface --> [hostapd, -B, /etc/hostapd-wlan0.conf]
Nov 01 12:28:58 raspberrypi Kura[709]: Storing message on topic: $LOG/#account-name/#client-id/V1/3/3/dhcpd, priority: 7
Nov 01 12:28:58 raspberrypi Kura[709]: Stored message on topic: $LOG/#account-name/#client-id/V1/3/3/dhcpd, priority: 7
Nov 01 12:28:58 raspberrypi Kura[709]: DataPublisherService not connected
Nov 01 12:29:10 raspberrypi Kura[709]: bringing interface wlan0 up
Nov 01 12:29:11 raspberrypi dhcpd[6994]: receive_packet failed on wlan0: Network is down
Nov 01 12:29:11 raspberrypi Kura[709]: starting hostapd for the wlan0 interface --> [hostapd, -B, /etc/hostapd-wlan0.conf]
Nov 01 12:29:11 raspberrypi Kura[709]: Storing message on topic: $LOG/#account-name/#client-id/V1/0/3/kernel, priority: 7
Nov 01 12:29:11 raspberrypi Kura[709]: Stored message on topic: $LOG/#account-name/#client-id/V1/0/3/kernel, priority: 7
Nov 01 12:29:11 raspberrypi Kura[709]: DataPublisherService not connected
Nov 01 12:29:22 raspberrypi Kura[709]: bringing interface wlan0 up
Nov 01 12:29:23 raspberrypi dhcpd[6994]: receive_packet failed on wlan0: Network is down
Nov 01 12:29:23 raspberrypi Kura[709]: starting hostapd for the wlan0 interface --> [hostapd, -B, /etc/hostapd-wlan0.conf]
Nov 01 12:29:35 raspberrypi Kura[709]: bringing interface wlan0 up
Nov 01 12:29:36 raspberrypi dhcpd[6994]: receive_packet failed on wlan0: Network is down
Nov 01 12:29:36 raspberrypi Kura[709]: starting hostapd for the wlan0 interface --> [hostapd, -B, /etc/hostapd-wlan0.conf]
Nov 01 12:29:48 raspberrypi Kura[709]: bringing interface wlan0 up
Nov 01 12:29:48 raspberrypi dhcpd[6994]: receive_packet failed on wlan0: Network is down
Nov 01 12:29:49 raspberrypi Kura[709]: starting hostapd for the wlan0 interface --> [hostapd, -B, /etc/hostapd-wlan0.conf]
Nov 01 12:30:01 raspberrypi Kura[709]: bringing interface wlan0 up
Nov 01 12:30:01 raspberrypi dhcpd[6994]: receive_packet failed on wlan0: Network is down
Nov 01 12:30:01 raspberrypi Kura[709]: starting hostapd for the wlan0 interface --> [hostapd, -B, /etc/hostapd-wlan0.conf]
Nov 01 12:30:02 raspberrypi Kura[709]: Storing message on topic: $LOG/#account-name/#client-id/V1/3/3/dhcpd, priority: 7
Nov 01 12:30:02 raspberrypi Kura[709]: Stored message on topic: $LOG/#account-name/#client-id/V1/3/3/dhcpd, priority: 7
Nov 01 12:30:02 raspberrypi Kura[709]: DataPublisherService not connected
Nov 01 12:30:13 raspberrypi Kura[709]: bringing interface wlan0 up
Nov 01 12:30:14 raspberrypi dhcpd[6994]: receive_packet failed on wlan0: Network is down
Nov 01 12:30:14 raspberrypi Kura[709]: starting hostapd for the wlan0 interface --> [hostapd, -B, /etc/hostapd-wlan0.conf]
Nov 01 12:30:14 raspberrypi Kura[709]: Storing message on topic: $LOG/#account-name/#client-id/V1/0/3/kernel, priority: 7
Nov 01 12:30:14 raspberrypi Kura[709]: Stored message on topic: $LOG/#account-name/#client-id/V1/0/3/kernel, priority: 7
Nov 01 12:30:14 raspberrypi Kura[709]: DataPublisherService not connected
Nov 01 12:30:26 raspberrypi Kura[709]: bringing interface wlan0 up
Nov 01 12:30:26 raspberrypi dhcpd[6994]: receive_packet failed on wlan0: Network is down
Nov 01 12:30:27 raspberrypi Kura[709]: starting hostapd for the wlan0 interface --> [hostapd, -B, /etc/hostapd-wlan0.conf]
Nov 01 12:30:38 raspberrypi Kura[709]: bringing interface wlan0 up
Nov 01 12:30:39 raspberrypi dhcpd[6994]: receive_packet failed on wlan0: Network is down
Nov 01 12:30:39 raspberrypi Kura[709]: starting hostapd for the wlan0 interface --> [hostapd, -B, /etc/hostapd-wlan0.conf]
Nov 01 12:30:51 raspberrypi Kura[709]: bringing interface wlan0 up
Nov 01 12:30:52 raspberrypi dhcpd[6994]: receive_packet failed on wlan0: Network is down
Nov 01 12:30:52 raspberrypi Kura[709]: starting hostapd for the wlan0 interface --> [hostapd, -B, /etc/hostapd-wlan0.conf]
Nov 01 12:31:04 raspberrypi Kura[709]: bringing interface wlan0 up
Nov 01 12:31:05 raspberrypi dhcpd[6994]: receive_packet failed on wlan0: Network is down
Nov 01 12:31:05 raspberrypi Kura[709]: starting hostapd for the wlan0 interface --> [hostapd, -B, /etc/hostapd-wlan0.conf]
Nov 01 12:31:05 raspberrypi Kura[709]: Storing message on topic: $LOG/#account-name/#client-id/V1/3/3/dhcpd, priority: 7
Nov 01 12:31:05 raspberrypi Kura[709]: Stored message on topic: $LOG/#account-name/#client-id/V1/3/3/dhcpd, priority: 7
Nov 01 12:31:05 raspberrypi Kura[709]: DataPublisherService not connected
Nov 01 12:31:16 raspberrypi Kura[709]: bringing interface wlan0 up
Nov 01 12:31:17 raspberrypi dhcpd[6994]: receive_packet failed on wlan0: Network is down
Nov 01 12:31:17 raspberrypi Kura[709]: starting hostapd for the wlan0 interface --> [hostapd, -B, /etc/hostapd-wlan0.conf]
Nov 01 12:31:18 raspberrypi Kura[709]: Storing message on topic: $LOG/#account-name/#client-id/V1/0/3/kernel, priority: 7
Nov 01 12:31:18 raspberrypi Kura[709]: Stored message on topic: $LOG/#account-name/#client-id/V1/0/3/kernel, priority: 7
Nov 01 12:31:18 raspberrypi Kura[709]: DataPublisherService not connected
Nov 01 12:31:29 raspberrypi Kura[709]: bringing interface wlan0 up
Nov 01 12:31:30 raspberrypi dhcpd[6994]: receive_packet failed on wlan0: Network is down
Nov 01 12:31:30 raspberrypi Kura[709]: starting hostapd for the wlan0 interface --> [hostapd, -B, /etc/hostapd-wlan0.conf]
```

MMaiero commented:
To notice that even if ac is selected, the UI at refresh keeps selecting a instead.

salvatore-coppola committed:
2.4 Ghz is disabled for A mode.

salvatore-coppola committed:
Added first draft of new kura wifi properties.

salvatore-coppola committed:
Now capabilites are extracted from IW commands.
Added some tests.

salvatore-coppola committed:
Code Cleanup

salvatore-coppola committed:
Removed SystemService dependency from NetworkAdminService

salvatore-coppola committed:
Removed ACS.
Fixed dfs and vht regex.
Fixed Wireless UI.
Fixed Radio Mode tooltip.
Fixed HostapdConfigTest

salvatore-coppola committed:
Removed leftover.

salvatore-coppola committed:
Fixed security Hotspots on regex: https://sonarcloud.io/project/security_hotspots?id=org.eclipse.kura%3Akura&branch=PR-3661&resolved=false&types=SECURITY_HOTSPOT

MMaiero commented:
Hello,
I have just tested this PR and found the following issues:
1. a new selection of the available SSID causes the password placeholder to be set in the wpa_supplicant config
2. when opening the page with a selected SSID in station mode (5ghz) the list of available channels is not only graded out (expected) but also empty, causing the configuration to not be complete so that the apply button doesn't apply new changes until the user selects a new SSID from the list

salvatore-coppola committed:
Configuration Loading is done in the proper way

salvatore-coppola committed:
Fixed bug in WpaSupplicantWriter.
Now channel is properly set in station mode.

salvatore-coppola commented:
> 
> 
> Hello, I have just tested this PR and found the following issues:
> 
>     1. a new selection of the available SSID causes the password placeholder to be set in the wpa_supplicant config
> 
>     2. when opening the page with a selected SSID in station mode (5ghz) the list of available channels is not only graded out (expected) but also empty, causing the configuration to not be complete so that the apply button doesn't apply new changes until the user selects a new SSID from the list

Theese 2 problems should be fixed with the last commit.

MMaiero commented:
I have found a possible bug in my setup.
I have set the access point mode in my wi-fi interface.
The UI allows to select different options between n/g/b, g/b, b and a. But, when I select one, it goes straight back to g/b.

MMaiero commented:
Plus I have the log filled with those messages here:

```
Nov 16 12:55:48 boltgate2031-64 ESF[1211]: {entrypoint=WebConsole, identity=admin, ip=172.16.0.5, web.path=/admin/denali/network, rpc.method=GwtNetworkServiceImpl.isIEEE80211ACSupported, session.id=node01dfhdwhhg61it1v3w9bm33ebpa0} UI Network - Success - Verify ieee80211ac is supported
Nov 16 12:55:48 boltgate2031-64 ESF[1211]: {entrypoint=WebConsole, identity=admin, ip=172.16.0.5, web.path=/admin/denali/network, rpc.method=GwtNetworkServiceImpl.findFrequencies, session.id=node01dfhdwhhg61it1v3w9bm33ebpa0} UI Network - Success - Get Wifi channels and frequencies
Nov 16 12:55:48 boltgate2031-64 ESF[1211]: {entrypoint=WebConsole, identity=admin, ip=172.16.0.5, web.path=/admin/denali/network, rpc.method=GwtNetworkServiceImpl.isIEEE80211ACSupported, session.id=node01dfhdwhhg61it1v3w9bm33ebpa0} UI Network - Success - Verify ieee80211ac is supported
Nov 16 12:55:48 boltgate2031-64 ESF[1211]: {entrypoint=WebConsole, identity=admin, ip=172.16.0.5, web.path=/admin/denali/network, rpc.method=GwtNetworkServiceImpl.findFrequencies, session.id=node01dfhdwhhg61it1v3w9bm33ebpa0} UI Network - Success - Get Wifi channels and frequencies
Nov 16 12:55:48 boltgate2031-64 ESF[1211]: {entrypoint=WebConsole, identity=admin, ip=172.16.0.5, web.path=/admin/denali/network, rpc.method=GwtNetworkServiceImpl.isIEEE80211ACSupported, session.id=node01dfhdwhhg61it1v3w9bm33ebpa0} UI Network - Success - Verify ieee80211ac is supported
Nov 16 12:55:48 boltgate2031-64 ESF[1211]: {entrypoint=WebConsole, identity=admin, ip=172.16.0.5, web.path=/admin/denali/network, rpc.method=GwtNetworkServiceImpl.findFrequencies, session.id=node01dfhdwhhg61it1v3w9bm33ebpa0} UI Network - Success - Get Wifi channels and frequencies
Nov 16 12:55:48 boltgate2031-64 ESF[1211]: {entrypoint=WebConsole, identity=admin, ip=172.16.0.5, web.path=/admin/denali/network, rpc.method=GwtNetworkServiceImpl.isIEEE80211ACSupported, session.id=node01dfhdwhhg61it1v3w9bm33ebpa0} UI Network - Success - Verify ieee80211ac is supported
Nov 16 12:55:48 boltgate2031-64 ESF[1211]: {entrypoint=WebConsole, identity=admin, ip=172.16.0.5, web.path=/admin/denali/network, rpc.method=GwtNetworkServiceImpl.findFrequencies, session.id=node01dfhdwhhg61it1v3w9bm33ebpa0} UI Network - Success - Get Wifi channels and frequencies
Nov 16 12:55:49 boltgate2031-64 ESF[1211]: {entrypoint=WebConsole, identity=admin, ip=172.16.0.5, web.path=/admin/denali/network, rpc.method=GwtNetworkServiceImpl.isIEEE80211ACSupported, session.id=node01dfhdwhhg61it1v3w9bm33ebpa0} UI Network - Success - Verify ieee80211ac is supported
Nov 16 12:55:49 boltgate2031-64 ESF[1211]: {entrypoint=WebConsole, identity=admin, ip=172.16.0.5, web.path=/admin/denali/network, rpc.method=GwtNetworkServiceImpl.findFrequencies, session.id=node01dfhdwhhg61it1v3w9bm33ebpa0} UI Network - Success - Get Wifi channels and frequencies
Nov 16 12:55:49 boltgate2031-64 ESF[1211]: {entrypoint=WebConsole, identity=admin, ip=172.16.0.5, web.path=/admin/denali/network, rpc.method=GwtNetworkServiceImpl.isIEEE80211ACSupported, session.id=node01dfhdwhhg61it1v3w9bm33ebpa0} UI Network - Success - Verify ieee80211ac is supported
Nov 16 12:55:49 boltgate2031-64 ESF[1211]: {entrypoint=WebConsole, identity=admin, ip=172.16.0.5, web.path=/admin/denali/network, rpc.method=GwtNetworkServiceImpl.findFrequencies, session.id=node01dfhdwhhg61it1v3w9bm33ebpa0} UI Network - Success - Get Wifi channels and frequencies
Nov 16 12:55:49 boltgate2031-64 ESF[1211]: {entrypoint=WebConsole, identity=admin, ip=172.16.0.5, web.path=/admin/denali/network, rpc.method=GwtNetworkServiceImpl.isIEEE80211ACSupported, session.id=node01dfhdwhhg61it1v3w9bm33ebpa0} UI Network - Success - Verify ieee80211ac is supported
Nov 16 12:55:49 boltgate2031-64 ESF[1211]: {entrypoint=WebConsole, identity=admin, ip=172.16.0.5, web.path=/admin/denali/network, rpc.method=GwtNetworkServiceImpl.isIEEE80211ACSupported, session.id=node01dfhdwhhg61it1v3w9bm33ebpa0} UI Network - Success - Verify ieee80211ac is supported
Nov 16 12:55:49 boltgate2031-64 ESF[1211]: {entrypoint=WebConsole, identity=admin, ip=172.16.0.5, web.path=/admin/denali/network, rpc.method=GwtNetworkServiceImpl.findFrequencies, session.id=node01dfhdwhhg61it1v3w9bm33ebpa0} UI Network - Success - Get Wifi channels and frequencies
Nov 16 12:55:49 boltgate2031-64 ESF[1211]: {entrypoint=WebConsole, identity=admin, ip=172.16.0.5, web.path=/admin/denali/network, rpc.method=GwtNetworkServiceImpl.findFrequencies, session.id=node01dfhdwhhg61it1v3w9bm33ebpa0} UI Network - Success - Get Wifi channels and frequencies
Nov 16 12:55:50 boltgate2031-64 ESF[1211]: {entrypoint=WebConsole, identity=admin, ip=172.16.0.5, web.path=/admin/denali/network, rpc.method=GwtNetworkServiceImpl.isIEEE80211ACSupported, session.id=node01dfhdwhhg61it1v3w9bm33ebpa0} UI Network - Success - Verify ieee80211ac is supported
Nov 16 12:55:50 boltgate2031-64 ESF[1211]: {entrypoint=WebConsole, identity=admin, ip=172.16.0.5, web.path=/admin/denali/network, rpc.method=GwtNetworkServiceImpl.findFrequencies, session.id=node01dfhdwhhg61it1v3w9bm33ebpa0} UI Network - Success - Get Wifi channels and frequencies
Nov 16 12:55:50 boltgate2031-64 ESF[1211]: {entrypoint=WebConsole, identity=admin, ip=172.16.0.5, web.path=/admin/denali/network, rpc.method=GwtNetworkServiceImpl.isIEEE80211ACSupported, session.id=node01dfhdwhhg61it1v3w9bm33ebpa0} UI Network - Success - Verify ieee80211ac is supported
Nov 16 12:55:50 boltgate2031-64 ESF[1211]: {entrypoint=WebConsole, identity=admin, ip=172.16.0.5, web.path=/admin/denali/network, rpc.method=GwtNetworkServiceImpl.findFrequencies, session.id=node01dfhdwhhg61it1v3w9bm33ebpa0} UI Network - Success - Get Wifi channels and frequencies
Nov 16 12:55:50 boltgate2031-64 ESF[1211]: {entrypoint=WebConsole, identity=admin, ip=172.16.0.5, web.path=/admin/denali/network, rpc.method=GwtNetworkServiceImpl.isIEEE80211ACSupported, session.id=node01dfhdwhhg61it1v3w9bm33ebpa0} UI Network - Success - Verify ieee80211ac is supported
Nov 16 12:55:50 boltgate2031-64 ESF[1211]: {entrypoint=WebConsole, identity=admin, ip=172.16.0.5, web.path=/admin/denali/network, rpc.method=GwtNetworkServiceImpl.findFrequencies, session.id=node01dfhdwhhg61it1v3w9bm33ebpa0} UI Network - Success - Get Wifi channels and frequencies
Nov 16 12:55:51 boltgate2031-64 ESF[1211]: {entrypoint=WebConsole, identity=admin, ip=172.16.0.5, web.path=/admin/denali/network, rpc.method=GwtNetworkServiceImpl.isIEEE80211ACSupported, session.id=node01dfhdwhhg61it1v3w9bm33ebpa0} UI Network - Success - Verify ieee80211ac is supported
Nov 16 12:55:51 boltgate2031-64 ESF[1211]: {entrypoint=WebConsole, identity=admin, ip=172.16.0.5, web.path=/admin/denali/network, rpc.method=GwtNetworkServiceImpl.findFrequencies, session.id=node01dfhdwhhg61it1v3w9bm33ebpa0} UI Network - Success - Get Wifi channels and frequencies
Nov 16 12:55:58 boltgate2031-64 ESF[1211]: {entrypoint=WebConsole, identity=admin, ip=172.16.0.5, web.path=/admin/denali/network, rpc.method=GwtNetworkServiceImpl.isIEEE80211ACSupported, session.id=node01dfhdwhhg61it1v3w9bm33ebpa0} UI Network - Success - Verify ieee80211ac is supported
Nov 16 12:55:58 boltgate2031-64 ESF[1211]: {entrypoint=WebConsole, identity=admin, ip=172.16.0.5, web.path=/admin/denali/network, rpc.method=GwtNetworkServiceImpl.findFrequencies, session.id=node01dfhdwhhg61it1v3w9bm33ebpa0} UI Network - Success - Get Wifi channels and frequencies
Nov 16 12:56:00 boltgate2031-64 ESF[1211]: {entrypoint=WebConsole, identity=admin, ip=172.16.0.5, web.path=/admin/denali/network, rpc.method=GwtNetworkServiceImpl.isIEEE80211ACSupported, session.id=node01dfhdwhhg61it1v3w9bm33ebpa0} UI Network - Success - Verify ieee80211ac is supported
Nov 16 12:56:00 boltgate2031-64 ESF[1211]: {entrypoint=WebConsole, identity=admin, ip=172.16.0.5, web.path=/admin/denali/network, rpc.method=GwtNetworkServiceImpl.findFrequencies, session.id=node01dfhdwhhg61it1v3w9bm33ebpa0} UI Network - Success - Get Wifi channels and frequencies
```

I understand the purpose for debug, but seem to be too many for production use.

salvatore-coppola committed:
Removed Audit on isIEEE80211ACSupported.
Fixed bug on radio mode dropdown list.

salvatore-coppola committed:
Removed useless field.
